packages/client/bin/repl.ts

```diff
@@ -0,0 +1,85 @@
+import process from 'process'
+import repl from 'repl'
+
+const setupClient = async () => {
+  const { readFileSync } = await import('fs')
+
+  const { createCommonFromGethGenesis } = await import('@ethereumjs/common')
+  const { createInlineClient } = await import('../test/sim/simutils.ts')
+  const { Config } = await import('../src/config.ts')
+  const { getLogger } = await import('../src/logging.ts')
+  const { startRPCServers } = await import('./startRPC.ts')
+  const genesisFile = JSON.parse(readFileSync('./bin/genesis.json', 'utf-8'))
+  const chainName = 'pectra'
+  const common = createCommonFromGethGenesis(genesisFile, {
+    chain: chainName,
+  })
+  const config = new Config({
+    common,
+    logger: getLogger({ logLevel: 'info' }),
+    saveReceipts: true,
+    enableSnapSync: true,
+  })
+  const client = await createInlineClient(config, common, {}, '', true)
+  const servers = startRPCServers(client, {
+    rpc: true,
+    rpcAddr: '0.0.0.0',
+    rpcPort: 8545,
+    ws: false,
+    wsPort: 0,
+    wsAddr: '0.0.0.0',
+    rpcEngine: true,
+    rpcEngineAddr: '0.0.0.0',
+    rpcEnginePort: 8551,
+    wsEngineAddr: '0.0.0.0',
+    wsEnginePort: 8552,
+    rpcDebug: 'eth',
+    rpcDebugVerbose: 'false',
+    helpRPC: false,
+    jwtSecret: '',
+    rpcEngineAuth: false,
+    rpcCors: '',
+  })
+  return { client, executionRpc: servers[0], engineRpc: servers[1] }
+}
+
+const setupRepl = async () => {
+  const { client, executionRpc, engineRpc } = await setupClient()
+
+  const replServer = repl.start({
+    prompt: 'EthJS > ',
+    ignoreUndefined: true,
+  })
+
+  // bootstrap contexts or modules
```

Comment:
We could potentially make the rpc servers and the API's they serve available as contexts here, but on first look, this wouldn't be very usable. A more user friendly option might be defining commands in the repl that would interact with the client and rpc server objects that could be called like this: `.getBlock 0x01`.